### PR TITLE
[SYCL][L0] Improvements to immediate command list settings

### PR DESCRIPTION
**Pull request by jandres742**
- Cache the setting for immediate command list, instead of reading the environment variable on every query.
- Same for the setting for scope events.
- Ignore SYCL_PI_LEVEL_ZERO_DEVICE_SCOPE_EVENTS when immediate lists are used.
- Keep SYCL_PI_LEVEL_ZERO_DEVICE_SCOPE_EVENTS, as it may be useful for experiments in the future.

### CODE REVIEW

**Comment by jandres742** [comment]
@smaslov-intel , @rdeodhar , please check

File: sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero.hpp [hunk 1]
```diff
@@ -283,6 +299,8 @@ struct _ur_device_handle_t : _pi_object {
   // Return whether to use immediate commandlists for this device.
   ImmCmdlistMode useImmediateCommandLists();
 
+  EventsScope eventsScope = AllHostVisible;
```
Comment (smaslov-intel) [inline sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero.hpp:302]: Coding style is to start var names with an upper case and functions names with a lower case.

```suggestion
  EventsScope EventsScope = AllHostVisible;

```

Also please add a comment about what this new member does.

File: sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero.hpp [hunk 2]
```diff
@@ -270,6 +285,7 @@ struct _ur_device_handle_t : _pi_object {
   // Whether to use immediate commandlists for queues on this device.
   // For some devices (e.g. PVC) immediate commandlists are preferred.
   bool ImmCommandListsPreferred;
+  bool isImmedidateCommandListUsed = false;
```
Comment (smaslov-intel) [inline sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero.hpp:288]: I suggest that we fuse these 2 booleans into just one.

Comment (jandres742) [reply sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero.hpp:288]: thanks @smaslov-intel . good idea. changing it.

File: sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero.hpp (hunk 1 above)
Comment (jandres742) [reply sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero.hpp:302]: @smaslov-intel : cool, adding it.

**Review by smaslov-intel — APPROVED** [review]

**Review by bader — APPROVED** [review]
sycl/doc/EnvironmentVariables.md LGTM.

**Review by againull — APPROVED** [review]

**Pull request merged**